bottlecap/src/traces/trace_processor.rs

```diff
@@ -101,6 +106,10 @@ fn filter_span_from_lambda_library_or_runtime(span: &Span) -> bool {
         }
     }
     if span.resource == INVOCATION_SPAN_RESOURCE {
+        let mut guard = context_buffer.lock().expect("lock poisoned");
+        if let Some(request_id) = span.meta.get("request_id") {
```

Comment:
We assume the top level span is including the `request_id`, allowing us to match the data in the context buffer.

This is done in Ruby and Go.

.NET would have to send the top-level span first, as that one gets ditched and never arrives, should be a quick fix, but might need to get access to context so we can tag the current span with the `request_id`.

Not sure how Java handles this.